src/Pulsar.Client/Auth/AuthenticationDataTls.fs

```diff
@@ -0,0 +1,34 @@
+﻿namespace Pulsar.Client.Auth
+
+open System.IO
+open System.Security.Cryptography.X509Certificates
+open System.Security.Cryptography.X509Certificates
+open Pulsar.Client.Api
+
+type internal AuthenticationDataTls(certFile: FileInfo) =
+    inherit AuthenticationDataProvider()
+    
+    let createCertCollection() =
+        use cert = new X509Certificate2(certFile.FullName)
+        if cert.HasPrivateKey then
+            let result  = X509CertificateCollection()
+            result.Add(new X509Certificate(certFile.FullName)) |> ignore
+            result
+        else
+            X509CertificateCollection()

```

Comment:
Why do we return empty collection for this case? It seems that throwing an exception "Certificate doesn't contain a private key" is a more user-friendly way to handle it

Reply:
Fixed